core/server/master/src/main/java/alluxio/master/meta/checkconf/ServerConfigurationChecker.java

```diff
@@ -0,0 +1,251 @@
+          confErrors.putIfAbsent(scope, new ArrayList<>());
+          confErrors.get(scope).add(wrongProperty);
+        } else {
+          confWarns.putIfAbsent(scope, new ArrayList<>());
+          confWarns.get(scope).add(wrongProperty);
+        }
+      }
+    }
+
+    // Update configuration status
+    Map<Scope, Status> configStatus = new HashMap<>();
+    Status masterStatus = Status.PASSED;
+    Status workerStatus = Status.PASSED;
+
+    for (Map.Entry<Scope, List<WrongProperty>> entry : confWarns.entrySet()) {
+      if (entry.getValue().size() > 0) {
+        Scope scope = entry.getKey();
+        if (scope.contains(Scope.MASTER)) {
+          masterStatus = Status.WARN;
+        }
+        if (scope.contains(Scope.WORKER)) {
+          workerStatus = Status.WARN;
+        }
+      }
+    }
+    for (Map.Entry<Scope, List<WrongProperty>> entry : confErrors.entrySet()) {
+      if (entry.getValue().size() > 0) {
+        Scope scope = entry.getKey();
+        if (scope.contains(Scope.MASTER)) {
+          masterStatus = Status.FAILED;
+        }
+        if (scope.contains(Scope.WORKER)) {
+          workerStatus = Status.FAILED;
+        }
+      }
+    }
+
+    configStatus.put(Scope.MASTER, masterStatus);
+    configStatus.put(Scope.WORKER, workerStatus);
+
+    Status reportStatus =
+        (masterStatus.equals(Status.FAILED) || workerStatus.equals(Status.FAILED))
+        ? Status.FAILED : (masterStatus.equals(Status.WARN) || workerStatus.equals(Status.WARN))
+        ? Status.WARN : Status.PASSED;
+
+    mConfigCheckReport = new ConfigCheckReport(confErrors, confWarns, configStatus, reportStatus);
+  }
+
+  /**
+   * @return the configuration checker report
+   */
+  public synchronized ConfigCheckReport getConfigCheckReport() {
+    return mConfigCheckReport;
+  }
+
+  /**
+   * Generates the configuration map to find wrong configuration.
+   * The map is of format Map<PropertyKey, Map<Value, List<Host:Port>>.
+   *
+   * @return the generated configuration map
+   */
+  private Map<PropertyKey, Map<String, List<String>>> generateConfMap() {
+    Map<PropertyKey, Map<String, List<String>>> confMap = new HashMap<>();
+    fillConfMap(confMap, mMasterRecord.getConfMap());
+    fillConfMap(confMap, mWorkerRecord.getConfMap());
+    return confMap;
+  }
+
+  /**
+   * Fills the configuration map.
+   *
+   * @param targetMap the map to fill
+   * @param recordMap the map to get data from
+   */
+  private void fillConfMap(Map<PropertyKey, Map<String, List<String>>> targetMap,
+      Map<Address, List<ConfigRecord>> recordMap) {
+    for (Map.Entry<Address, List<ConfigRecord>> record : recordMap.entrySet()) {
+      Address address = record.getKey();
+      String addressStr = String.format("%s:%s", address.getHost(), address.getRpcPort());
+      for (ConfigRecord conf : record.getValue()) {
+        PropertyKey key = conf.getKey();
+        if (key.getConsistencyLevel().equals(ConsistencyCheckLevel.IGNORE)) {
```

Comment:
nit: for enum comparison, I would do 
`key.getConsistencyLevel() == ConsistencyCheckLevel.IGNORE` which also works for `null` case.

Reply:
Changed, thanks! 
Why will it also work for null?
IGNORE will transfer to null?

Reply:
I mean `key.getConsistencyLevel()` can be `null` . vs. in the previous pattern, when `key.getConsistencyLevel()` is `null`, NPE will be thrown